pkg/process/monitor/process_monitor.go

```diff
@@ -207,10 +234,14 @@ func (pm *ProcessMonitor) mainEventLoop() {
 		logTicker.Stop()
 		// Marking netlink to stop, so we won't get any new events.
 		close(pm.netlinkDoneChannel)
+
 		// waiting for the callbacks runners to finish
-		close(pm.callbackRunner)
```

Comment:
Shouldn't you be trying to prevent the sends to the channel instead of not closing it? If the channel happens to be full, you don't close it, and stop reading from it, then the sends could block which I imagine would cause other problems.

Reply:
One suggestion, introduce a channel/atomic indicating the monitor is "done", that you check before sending to `pm.callbackRunner`

Reply:
>  If the channel happens to be full, you don't close it, and stop reading from it, then the sends could block which I imagine would cause other problems

Your point is smart, and a good fix is to make the sending process non-blocking.

I aimed to adhere to `The Channel Closing Principle`, emphasizing not closing a channel when multiple concurrent senders are involved - an acceptable practice in Go.

Employing an atomic boolean/channel to indicate the monitor's completion could introduce a race condition. For instance, if the main goroutine is on the verge of writing to the `callbackRunner` channel (after checking the proposed atomic value) and a context switch occurs to the monitor goroutine, there's a chance it sets the boolean/channel to true and closes the channel. Subsequently, another context switch back to the main goroutine could result in an attempt to write to a closed channel.

This could be resolved by using the existing `processExecCallbacksMutex` before attempting to close the channel. However, I'm uncertain if this is a more elegant solution compared to not closing the channel, and there's also the concern about the performance overhead incurred by checking an atomic boolean value for each pid